### PR TITLE
Issue 5271 - relax httpcliient nocontent assertion

### LINKED ISSUE #5271: HTTP response 204 No Content should not crash http client if a Content-Length header is included
**Describe the bug (REQUIRED)**
The spec for 204 No Content states that no Content-Length header should be sent. The Defold http client is strict about this and will assert if a Content-Length header is included.

**To Reproduce (REQUIRED)**
1. Make a HTTP request to a resource which returns 204 No Content and a Conten-Length header
2. Observe crash

**Expected behavior (REQUIRED)**
The HTTP client should not crash (but perhaps log a warning)

**Defold version (REQUIRED):**
  - 1.2.174

**Platforms (REQUIRED):**
 - Platforms: All

**Additional context (OPTIONAL):**
Forum thread: https://forum.defold.com/t/application-crashes-on-http-request/66587

### PR DESCRIPTION
**Pull request by otsakir**
This is about #5271. This patch is trivial. I just removed the assertion regarding Conent-Length if http status 204 is returned. 

#### Manual tests

* I've reproduced the issue using a custom server that would _return content_ for 204 responses. I got the exception in the log.
* I removed assertion and tested again and verified that the reqeuest was indeed made and application didn't throw any error.
* My dev and target platform  was x86_64-linux.

#### Automated test

* I checked that nothing else had broken by running the unit tests under engine/dlib module. 
* I tried to build an automated test that would detect the bug and pass after the fix. I investigated engine/dlib/src/test/http_server/TestHttpServer.java and  tried modify but jetty refused to return response body if 204 status was defined. For some reason it enforces proper syntax in the response :-). Thoughts ?

### CODE REVIEW

File: engine/dlib/src/dlib/http_client.cpp [hunk 1]
```diff
@@ -924,7 +924,7 @@ if (sock_res != dmSocket::RESULT_OK)\
 
         if (response.m_Status == 204 /* No Content*/)
         {
-            assert(response.m_ContentLength == -1);
+            // assume content length is zero. No need to complain if an invalid response non empty content is received.
```
Comment (JCash) [inline engine/dlib/src/dlib/http_client.cpp:927]: Makes good sense to me!

**Review by JCash — APPROVED** [review]
Looks good!

File: engine/dlib/src/dlib/http_client.cpp (hunk 1 above)
Comment (britzl) [reply engine/dlib/src/dlib/http_client.cpp:927]: I agree!

Comment (otsakir) [reply engine/dlib/src/dlib/http_client.cpp:927]: :+1:

**Pull request merged**